map_model/src/pathfind/v2.rs

```diff
@@ -3,82 +3,181 @@
 //! things here will probably move into pathfind/mod.rs.
 
 use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use geom::Duration;
 
 use crate::pathfind::uber_turns::UberTurnV2;
-use crate::{DirectedRoadID, Map, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn};
+use crate::{
+    DirectedRoadID, Map, MovementID, Path, PathConstraints, PathRequest, PathStep, TurnID, UberTurn,
+};
+
+/// One step along a path.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum PathStepV2 {
+    /// Original direction
+    Along(DirectedRoadID),
+    /// Opposite direction, sidewalks only
+    Contraflow(DirectedRoadID),
+    Turn(MovementID),
+}
 
-/// Transform a sequence of roads representing a path into the current lane-based path, by picking
-/// particular lanes and turns to use.
-pub fn path_v2_to_v1(
+/// A path between two endpoints for a particular mode. This representation is immutable and doesn't
```

Comment:
I'm calling out the "immutable" part, because v1 `Path` has some complexity because it's used in the simulation as an agent executes the path. I don't `PathV2` should ever have that.